src/uct/ib/base/ib_md.c

```diff
@@ -1213,12 +1234,17 @@ static ucs_status_t uct_ib_query_md_resources(uct_component_t *component,
     }
 
     for (i = 0; i < num_devices; ++i) {
+        /* Skip non-existent and non-accessible devices */
+        if (!uct_ib_check_device_exists(device_list[i])) {
+            continue;
+        }
         ucs_snprintf_zero(resources[i].md_name, sizeof(resources[i].md_name),
```

Comment:
should it be something like `resources[num_resources].md_name`?

Reply:
A nice one! Thanks.